core/src/Instance/AadAuthority.cs

```diff
@@ -107,6 +108,13 @@ internal override string GetTenantId()
         {
             return GetFirstPathSegment(CanonicalAuthority);
         }
-        
+
+        internal override void UpdateTenantId(string tenantId)
+        {
+            Uri authorityUri = new Uri(CanonicalAuthority);
+
+            CanonicalAuthority = 
+                string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", authorityUri.Authority, tenantId);
```

Comment:
>"https://{0}/{1}/" [](start = 60, length = 18)

for b2c you made this a const... perhaps you should do the same here for consistency